pr_agent/git_providers/azuredevops_provider.py

```diff
@@ -324,6 +325,12 @@ def publish_comment(self, pr_comment: str, is_temporary: bool = False, thread_co
             )
 
     def publish_description(self, pr_title: str, pr_body: str):
+        if len(pr_body) > MAX_PR_DESCRIPTION_LENGTH:
+            trunction_message = " ... (description truncated due to length limit)" 
+            pr_body = pr_body[:MAX_PR_DESCRIPTION_LENGTH - len(trunction_message)] + trunction_message
+            get_logger().warning(
+                "PR description exceeds the maximum character limit of 4000. Truncating the description."
+            )
```

Comment:
this is far from optimal mechansim

when you clip an HTML table in the middle, the entire rendering can go bad.
much better logic:

if too long:
- remove usage guide
if still too long:
- remove walkthrough table
if still too long:
- brute force trunctation

Reply:
i see there is plenty of configurable options to change the description (e.g remove the usage guide), so writing this logic will complicate the generic flow.
what i can do instead is just if size >= configurable limit than force description as comment. what do you think? i can do it only for azure devops or for in the PRDescription donctroller

Reply:
the defaults are used 99.9% of the time.
and even if changed, than no harm done because there is the fallback.

the logic should be:

if too long:

remove entire usage guide
if still too long:
remove entire walkthrough table
if still too long:
brute force truncation